service/frontend/workflowHandler.go

```diff
@@ -455,6 +456,10 @@ func (wh *WorkflowHandler) StartWorkflowExecution(ctx context.Context, request *
 		return nil, wh.error(errRequestIDTooLong, scope)
 	}
 
+	searchattribute.SetType(
+		request.SearchAttributes,
+		searchattribute.ConvertDynamicConfigToIndexedValueTypes(wh.config.ValidSearchAttributes()))
+
```

Comment:
remove this?

Reply:
SA validator use the type from metadata now. So it must be set before running validation.